cypress/e2e/article.cy.js

```diff
@@ -1,24 +1,163 @@
 /// <reference types="cypress" />
 /// <reference types="../support" />
+import { faker } from '@faker-js/faker';
+import { generateNewArticle } from '../support/generateNewArticle';
 
-describe('Article', () => {
-  before(() => {
 
-  });
 
-  beforeEach(() => {
-    cy.task('db:clear');
-  });
+describe('Create Article page', () => {
 
-  it('should be created using New Article form', () => {
+   let user;
+   const errorMsg = 'Oops!';
+   const { title, description, body, tags } = generateNewArticle();
 
-  });
+   beforeEach(() => {
+      cy.task('db:clear');
+      cy.task('generateUser').then(generateUser => {
+         user = generateUser;
+      });
+      cy.register().then(user => {
+         cy.login(user).then(() => user);
+      }).as('user');
+      cy.visit(`#/editor`);
+   });
 
-  it('should be edited using Edit button', () => {
+   it('should have New article form', () => {
+      cy.findByPlaceholder('Article Title').should('exist');
+      cy.findByPlaceholder('What\'s this article about?').should('exist');
+      cy.findByPlaceholder('Write your article (in markdown)').should('exist');
+      cy.findByPlaceholder('Enter tags').should('exist');
+      cy.getByDataQa('publish-btn').should('exist');
+   });
 
-  });
+   it('should create an article with filled all fields', () => {
+      cy.findByPlaceholder('Article Title').type(title);
+      cy.findByPlaceholder('What\'s this article about?').type(description);
+      cy.findByPlaceholder('Write your article (in markdown)').type(body);
+      cy.findByPlaceholder('Enter tags').type(tags);
+      cy.getByDataQa('publish-btn').click();
+      cy.url(`articles/${title}-`);
+      cy.get('h1').should('contain.text', title);
+   });
 
-  it('should be deleted using Delete button', () => {
+   it('should create an article with empty Tags field', () => {
+      cy.findByPlaceholder('Article Title').type(title);
+      cy.findByPlaceholder('What\'s this article about?').type(description);
+      cy.findByPlaceholder('Write your article (in markdown)').type(body);
+      cy.getByDataQa('publish-btn').click();
+      cy.url(`articles/${title}-`);
+      cy.get('h1').should('contain.text', title);
+   });
 
-  });
+   it('should create an article with empty Article Description field', () => {
+      cy.findByPlaceholder('Article Title').type(title);
+      cy.findByPlaceholder('Write your article (in markdown)').type(body);
+      cy.findByPlaceholder('Enter tags').type(tags);
+      cy.getByDataQa('publish-btn').click();
+      cy.url(`articles/${title}-`);
```

Comment:
Replace 'cy.url(`articles/${title}-`);' with 'cy.url().should('contain', `articles/${title}-`);' to correctly check the URL.